radarr/movieeditor.go

```diff
@@ -13,17 +13,36 @@ const bpMovieEditor = bpMovie + "/editor"
 
 // BulkEdit is the input for the bulk movie editor endpoint.
 // You may use starr.True(), starr.False(), starr.Int64(), and starr.String() to add data to the struct members.
+// Use Availability.Ptr() to add a value to minimum availability, and starr.ApplyTags.Ptr() for apply tags.
 type BulkEdit struct {
-	MovieIDs            []int64 `json:"movieIds"`
-	Monitored           *bool   `json:"monitored,omitempty"`
-	QualityProfileID    *int64  `json:"qualityProfileId,omitempty"`
-	MinimumAvailability *string `json:"minimumAvailability,omitempty"` // tba
-	RootFolderPath      *string `json:"rootFolderPath,omitempty"`      // path
-	Tags                []int   `json:"tags,omitempty"`                // [0]
-	ApplyTags           *string `json:"applyTags,omitempty"`           // add
-	MoveFiles           *bool   `json:"moveFiles,omitempty"`
-	DeleteFiles         *bool   `json:"deleteFiles,omitempty"`        // delete only
-	AddImportExclusion  *bool   `json:"addImportExclusion,omitempty"` // delete only
+	MovieIDs            []int64          `json:"movieIds"`
+	Monitored           *bool            `json:"monitored,omitempty"`
+	QualityProfileID    *int64           `json:"qualityProfileId,omitempty"`
+	MinimumAvailability *Availability    `json:"minimumAvailability,omitempty"` // tba
+	RootFolderPath      *string          `json:"rootFolderPath,omitempty"`      // path
+	Tags                []int            `json:"tags,omitempty"`                // [0]
+	ApplyTags           *starr.ApplyTags `json:"applyTags,omitempty"`           // add
+	MoveFiles           *bool            `json:"moveFiles,omitempty"`
+	DeleteFiles         *bool            `json:"deleteFiles,omitempty"`        // delete only
+	AddImportExclusion  *bool            `json:"addImportExclusion,omitempty"` // delete only
+}
+
+// Availability is an enum used as MinimumAvailability in a few places throughout Radarr.
+type Availability string
+
+// Availability / MinimumAvailability constants.
+// https://radarr.video/docs/api/#/MovieEditor/put_api_v3_movie_editor
+const (
+	AvailabilityToBeAnnounced Availability = "tba"
+	AvailabilityAnnounced     Availability = "announced"
+	AvailabilityInCinemas     Availability = "inCinemas"
+	AvailabilityReleased      Availability = "released"
+	AvailabilityDeleted       Availability = "deleted"
+)
```

Comment:
this is nice for a user perspective since it provides users with valid values. However, users can still use any string they want... so I'd say it doesn't make much difference for me, if you like this approach you can continue to use it!

Reply:
It's all about users. This, if nothing else, gives them an easy way to see what values are appropriate.

Reply:
:(

![Screen Shot 2022-10-07 at 11 18 15 AM](https://user-images.githubusercontent.com/2402929/194623813-68519080-376e-4f38-8bd6-ac61dd6d51f1.png)

Reply:
Practical use would be the 3 nit mentioned

- deleted is a status from tmdb
- tba is legacy carryover 

neither really has any practical use

Reply:
All good. I just made them constants. Users don't have to use them. :)